### ISSUE TITLE
Will not crack wallet.aes.json with password known 

### ISSUE THREAD

**Issue by azjnson** [gurnec/btcrecover#192]
I created a new wallet on block chain and used the used the download-blockchain-wallet.py to create the wallet.aes.json file.
I then put my password in to the token list as 3 parts and ran the software and it failed to find the password.
Shouldn't this find the password?
See my post below for issues I was seeing on the all test script.
Thank you for your support.

**Comment by pedrohrcunha** [comment]
Try to put your pass and its variations separadetly but all in the same line.

**Comment by azjnson** [comment]
I put the password as one word just like I enter it on the web site and it still fails.

This is the command I am using is this correct?

C:\wallet\btcrecover-master>c:\python27\python btcrecover.py --wallet wallet.aes.json --tokenlist tokens.txt
Starting btcrecover 0.17.9 on Python 2.7.14 64-bit, 16-bit unicodes, 32-bit ints
Wallet difficulty: 5,000 PBKDF2-SHA1 iterations
Using 4 worker threads
1 of 1 [############################################################] 0:00:00,
Password search exhausted

C:\wallet\btcrecover-master>

**Comment by pedrohrcunha** [comment]
And in the website is working normally? You sure there are no typos? Caps? Lowercase? 

If this is the case, only @gurnec can help you out.

**Comment by azjnson** [comment]
I copied it directly..
the format is like
1234#Abcdefgr#4321
Could there be something in the formatting?
I tried it on two different computers and it is the same.
J

**Comment by pedrohrcunha** [comment]
You sure you're using the right .json file? The one really connected to this password? 

I'm out of alternatives, let's wait for some mod to help you out.

**Comment by azjnson** [comment]
Yes it it the correct .json file. I dragged it from the directory to the blockchain site and opened it with the password.
Hopefully someone can help me with this.....

**Comment by gurnec** [comment]
Sorry, but I can't reproduce this.

I signed up for a new Blockchain.info account using the password `1234#Abcdefgr#4321`, downloaded the `wallet.aes.json` file using the `download-blockchain-wallet.py` script and the wallet identifier, created a token file containing only the password, and ran btcrecover and received the correct results:

```
Starting btcrecover 0.17.9 on Python 2.7.14 64-bit, 16-bit unicodes, 32-bit ints
Wallet difficulty: 5,000 PBKDF2-SHA1 iterations
Using 4 worker threads
0 of 1 [----------------------------------------------] 0:00:00, ETA:  --:--:--
Password found: '1234#Abcdefgr#4321'
```

Do you have a pound/hash `#` symbol at the *beginning* of your password or of any tokens? A `#` symbol introduces an ignored comment if (and only if) it is the *very first* character on a line, e.g. this tokens file:
```
1234
#Abcdefgr
#4321
```
only has one token, and two ignored comments in it. See the third paragraph [here for more details](https://github.com/gurnec/btcrecover/blob/master/docs/Limitations_and_Caveats.md#delimiters-spaces-and-special-symbols-in-passwords).

If that's not it, then I'm not sure what's going on. Did you do anything differently than the steps I did above when you tested? You said that

> the format is like 1234#Abcdefgr#4321

Is that the actual password you used, or was it something else?

**Comment by azjnson** [comment]
The steps you out lined above are what I did but it would not find the password.
So I created a new wallet on the Blockchain.info site with the same password as the last one.
I then did the exact same procedure and it worked with this new wallet.

Next I used the script to again download the wallet.aes.json file of the wallet that the password could not be found. I tried the exact same token file and it will not find the password of this .json file.

So for some reason it will not work on this one file.

Finally the question about the password I used.
the password is not exactly 1234#Abcdefgr#4321

It is 4 numbers then # followed by a Capitalized word then # then 4 numbers.

If you like the account was using to test the software is a burner so I could send you more precise info for you to try. 
I just don't want to post it here for everyone to see.
Thanks,
J

**Comment by pedrohrcunha** [comment]
@azjnson I MIGHT be in the same situation. I know my pass. I have the .json. And it doesn't work. But it doesn't work in blockchain.info login too.

**Comment by stev74** [comment]
I have the feeling there is some bug in the download script.
I have one blockchain wallet (from 2015) with a known password (I can login) but after downloading it with the script, btcrecover will not work.
On another, new created wallet it is working.
Is there any other script out there to download a blockchain-wallet?

**Comment by rterwedo** [comment]
@gurnec it depends what version the wallet is.  We have seen known issues with old .info wallets because it looks for an identifier that is not in old ones (hence false neg)

**Comment by anhday22** [comment]
Currently I have new technology to help open my wallet with a high success rate. If any of your wallet hasn't been successfully opened, please send it to me. I will try to open them.
http://recoverywalletcrypto.com
www.fb.me/recoverywallet
Thank

**Comment by nori19870522** [comment]
I am Japanese and do password analysis.
https://keychainx.io/ 

This is Robert's article.
https://news.bitstarz.com/bitstarz-news-interview-robert-rhodin-keychainx-ceo  

I may be able to find the password you lost. Do you need help with your login?

**Comment by btcexchanging** [comment]
if you haven't recovered your wallet, i can help you recover your wallet.
I have recovered most of the common wallets and recovery phrases for many users
you can reach me here at my telegram account:  @walletrecover

**Comment by viladmir2022** [comment]
hi, if you still have wallet problem. I can recover your lost wallet.
I have powerful servers and excellent software that can find any password.
and my fee is only 15-25% of the recovered fund
here is my telegram id: @viladmir2022  and Gmail: walletsrecoveryservice@gmail.com
